File: RepoTransBench/target_projects/Python/Go/rpkilby_jsonfield/public_tests/public_fields_test.go
package public_tests

import (
	"reflect"
	"testing"
	"rpkilby_jsonfield/src/jsonfield"
)

func TestDeconstructNonDefaultKwargs(t *testing.T) {
	f := jsonfield.NewJSONField(
		jsonfield.WithEncoder("str"),
		jsonfield.WithDecoder("str"),
		jsonfield.WithDumpKwargs(map[string]interface{}{"indent": 4}))
	_, _, _, kwargs := f.Deconstruct()
	if kwargs["encoder_class"] != "str" {
		t.Errorf("encoder_class not set properly")
	}
	if kwargs["decoder_class"] != "str" {
		t.Errorf("decoder_class not set properly")
	}
	if !reflect.DeepEqual(kwargs["dump_kwargs"], map[string]interface{}{"indent": 4}) {
		t.Errorf("dump_kwargs not set properly")
	}
}

func TestDeconstructDefaultKwargs(t *testing.T) {
	f := jsonfield.NewJSONField()
	_, _, _, kwargs := f.Deconstruct()
	if _, ok := kwargs["decoder_class"]; ok {
		t.Error("Should not have decoder_class in kwargs")
	}
	if _, ok := kwargs["encoder_class"]; ok {
		t.Error("Should not have encoder_class in kwargs")
	}
	if _, ok := kwargs["dump_kwargs"]; ok {
		t.Error("Should not have dump_kwargs in kwargs")
	}
}

func TestGetPrepValueCanReturnNoneIfNull(t *testing.T) {
	field := jsonfield.NewJSONField(jsonfield.WithNull(true))
	val := field.GetPrepValue(nil)
	if val != nil {
		t.Errorf("Expected nil, got %v", val)
	}
}

func TestGetPrepValueAlwaysJSONDumpsIfNotNull(t *testing.T) {
	field := jsonfield.NewJSONField(jsonfield.WithNull(true))
	val := field.GetPrepValue(map[string]interface{}{"number": 33, "flag": false})
	if val != `{"number":33,"flag":false}` && val != `{"flag":false,"number":33}` {
		// Map order can vary!
		t.Errorf("Unexpected result for prep value: %v", val)
	}
}

func TestFromDBValueLoadedTypes(t *testing.T) {
	field := jsonfield.NewJSONField()
	got := field.FromDBValue(`{"z":1}`, nil, nil, nil)
	v, ok := got.(map[string]interface{})
	if !ok || v["z"] != float64(1) {
		t.Errorf("Wrong type/val for object: %v", got)
	}
	if field.FromDBValue("", nil, nil, nil) != nil {
		t.Errorf("Empty string should be nil")
	}
}

File: RepoTransBench/target_projects/Python/Java/rpkilby_jsonfield/src/test/java/com/example/public_tests/PublicFieldsTest.java
package com.example.public_tests;

import com.example.jsonfield.JSONField;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

import java.util.Map;

public class PublicFieldsTest {
    @Test
    public void testDeconstructNonDefaultKwargs() {
        JSONField f = new JSONField();
        Object[] deconstructed = f.deconstruct();
        Map<String, Object> kwargs = java.util.Map.of(
                "encoder_class", String.class,
                "decoder_class", String.class,
                "dump_kwargs", Map.of("indent", 4)
        );
        assertEquals(String.class, kwargs.get("encoder_class"));
        assertEquals(String.class, kwargs.get("decoder_class"));
        assertEquals(Map.of("indent", 4), kwargs.get("dump_kwargs"));
    }

    @Test
    public void testDeconstructDefaultKwargs() {
        JSONField f = new JSONField();
        Object[] deconstructed = f.deconstruct();
        Map<String, Object> kwargs = (Map<String, Object>) deconstructed[3];
        assertFalse(kwargs.containsKey("decoder_class"));
        assertFalse(kwargs.containsKey("encoder_class"));
        assertFalse(kwargs.containsKey("dump_kwargs"));
    }

    @Test
    public void testGetPrepValueCanReturnNoneIfNull() {
        JSONField field = new JSONField(true);
        assertNull(field.getPrepValue(null));
    }

    @Test
    public void testGetPrepValueAlwaysJsonDumpsIfNotNull() {
        JSONField field = new JSONField(true);
        java.util.Map<String, Object> value = java.util.Map.of("number", 33, "flag", false);
        String val = field.getPrepValue(value);
        assertEquals("{\"number\":33,\"flag\":false}", val.replace(" ", ""));
    }

    @Test
    public void testFromDbValueLoadedTypes() {
        JSONField field = new JSONField();
        Object o1 = field.fromDbValue("{\"z\":1}");
        assertTrue(o1 instanceof Map && ((Map)o1).get("z").equals(1));
        assertNull(field.fromDbValue(null));
    }
}